src/arrayMethodSort.js

```diff
@@ -3,9 +3,39 @@
 /**
  * Implement method Sort
  */
+const defaultCompareFunction = (a, b) => {
+  if (a.toString() < b.toString()) {
+    return -1;
+  }
+
+  if (a.toString() > b.toString()) {
+    return 1;
+  }
+
+  return 0;
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function(compareFunction = defaultCompareFunction) {
     // write code here
+    for (let i = 0; i < this.length - 1; i++) {
+      let condition = 1;
+
+      if (i >= 0) {
+        condition = compareFunction(this[i + 1], this[i]);
+      }
+
+      if (condition < 0 && i >= 0) {
+        const temporaryContainer = this[i];
+
+        this[i] = this[i + 1];
+        this[i + 1] = temporaryContainer;
+
+        i -= 2;
+      }
+    }
```

Comment:
i'd refactor the function a bit
```suggestion
    for (let i = 0; i < this.length - 1; i++) {
      if (i < 0) {
        continue;
      }
      
      const condition = compareFunction(this[i + 1], this[i]);;

      if (condition < 0) {
        [this[i], this[i + 1]] = [this[i + 1], this[i]];

        i -= 2;
      }
    }
```

Reply:
Really, your variant is less complicated. Thanks for review